gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobPropCreator.java

```diff
@@ -105,12 +105,7 @@ protected MRCompactorJobPropCreator(Builder<?> builder) {
       LOG.warn("Input folder " + this.topicInputDir + " does not exist. Skipping topic " + topic);
       return emptyJobProps;
     }
-    Optional<State> jobProps = createJobProps(this.topicInputDir, this.topicOutputDir, this.topicTmpDir);
-    if (jobProps.isPresent()) {
-      return Collections.singletonList(jobProps.get());
-    } else {
-      return emptyJobProps;
-    }
+    return Collections.singletonList(createJobProps(this.topicInputDir, this.topicOutputDir, this.topicTmpDir));
```

Comment:
Before: `createJobProps` was responsible for determining whether a folder should be compacted. If so, return a job prop file, if not, return `Optional.absent()`. Now this responsibility is moved to `shouldProcessFolder`. `createJobProps` is only called if the folder should be compacted.